### PR TITLE
Add output of "ceph report"

### PR DESCRIPTION
**Pull request by toabctl**
Can be very useful when debugging Ceph problems.

### CODE REVIEW

**Comment by toabctl** [comment]
FYI ```ceph report``` produces quite some lines. for instance with 3 mons and no osds, 14420 lines.

File: susecloud [hunk 1]
```diff
@@ -285,6 +285,16 @@ else
 fi
 
 #############################################################
+section_header "Ceph report"
+
+if [ -e /usr/bin/ceph ]; then
+    ceph_report_dir=$(mktemp -d)
```
Comment (vuntz) [inline susecloud]: I know @aspiers wasn't keen on using /var/log in https://github.com/SUSE-Cloud/supportutils-plugin-susecloud/pull/33 but I actually feel it's the best place to put this, as `split-supportconfig` will then extract the log in the right place.

Or should we create a /var/log/supportconfig directory for this kind of "temporary" files?

Comment (aspiers) [inline susecloud]: Actually I was more objecting to creating something in `/var/log` and then deleting it shortly after.  But that's a very good point about `split-supportconfig`, so yes maybe it's worth either tolerating it or dumping in `/var/log/supportconfig`.

Comment (toabctl) [inline susecloud]: @vuntz , @aspiers : I added in https://github.com/toabctl/supportutils-plugin-susecloud/commit/911f2cc4d9500ed25ae5c37c15922ec40c877bd2 ```/var/log/supportconfig``` for now. And also added a timestamp to the file.

**Comment by toabctl** [comment]
@tserong : @osynge told me that you are currently developing something similar. Is there already a supportconfig plugin for the Storage product in the 1.0 release? If so I guess we can skip this PR and just install the plugin from the poduct media.

**Comment by tserong** [comment]
I've got https://github.com/tserong/supportutils-plugin-ses, but that happened too late to land on the Storage media.

File: susecloud [hunk 2]
```diff
@@ -285,6 +289,16 @@ else
 fi
 
 #############################################################
+section_header "Ceph report"
+
+if [ -e /usr/bin/ceph ]; then
+    now=$(date +%s)
+    ceph_report_name=$TMPLOGS_DIR/ceph_report_${now}.txt
+    ceph report > $ceph_report_name
```
Comment (mjura) [inline susecloud]: Some timeout would be helpful, because ceph command can hang sometimes.

Comment (mjura) [inline susecloud]: Redirecting stderr would be also nice, look on https://github.com/SUSE-Cloud/supportutils-plugin-suse-openstack-cloud/pull/47/files

**Comment by mjura** [comment]
I will check how `ceph report` looks like, maybe I will have more comments

**Comment by mjura** [comment]
Ok, I have checked what `ceph report` does. It makes full report status of cluster in really ugly dump.
I would be interested to have output from following commands instead `of ceph report`
```
ceph status
ceph auth list
ceph osd tree
ceph osd dump
```
In this way we will produce only 100 lines of logs but they will be much more precious for OpenStack integration.

**Comment by aspiers** [comment]
@mjura @tserong @toabctl What's the latest on this?

**Comment by tserong** [comment]
supportutils-plugin-ses currently does this:

````
ceph -s
ceph mon dump
ceph osd tree
ceph report
rados df
rbd ls
````

...so provided supportutils-plugin-ses is being installed on storage nodes deployed by Crowbar in SOC, that should be pretty good (although not identical to what @mjura was after).

**Comment by toabctl** [comment]
I think we should (if needed) improve supportutils-plugin-ses and make sure to install it on our storage nodes.

**Comment by toabctl** [comment]
According to https://github.com/SUSE-Cloud/supportutils-plugin-suse-openstack-cloud/pull/34#issuecomment-243730061 , the SES supportconfig plugin already collects the ```ceph report``` output. Closing...

**Pull request closed without merging**